aes-rng/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+#[repr(align(16))]
+#[derive(Clone)]
+pub struct AesCore {
+    round_keys: RoundKeys,
+    counter: M128,
+}
+
+impl AesCore {
+    /// Fill the given buffer with random data.
+    ///
+    /// Erases the key after filling the buffer.
+    #[inline]
+    pub fn fill(&mut self, buffer: &mut [u8]) {
+        let zero = M128::from((0, 0));
+        let one = M128::from((0, 1));
+        let two = M128::from((0, 2));
+        let mut c = [zero; 8];
+        let mut r = [zero; 8];
+        let mut s = [zero; 8];
+
+        c[0] = self.counter;
+        let mut remaining = buffer.len();
+        let mut buffer = buffer.as_mut_ptr();
+        while remaining > 128 {
+            c[1] = c[0] + one;
+            c[2] = c[0] + two;
+            c[3] = c[2] + one;
+            c[4] = c[2] + two;
+            c[5] = c[4] + one;
+            c[6] = c[4] + two;
+            c[7] = c[6] + one;
+            compute_rounds!(0, c, r, s, self.round_keys);
+            compute_rounds!(1, c, r, s, self.round_keys);
+            compute_rounds!(2, c, r, s, self.round_keys);
+            compute_rounds!(3, c, r, s, self.round_keys);
+            compute_rounds!(4, c, r, s, self.round_keys);
+            compute_rounds!(5, c, r, s, self.round_keys);
+            compute_rounds!(6, c, r, s, self.round_keys);
+            compute_rounds!(7, c, r, s, self.round_keys);
+            c[0] = c[7] + one;
+            unsafe {
+                r[0].store(buffer.offset(0));
+                r[1].store(buffer.offset(16));
+                r[2].store(buffer.offset(32));
+                r[3].store(buffer.offset(48));
+                r[4].store(buffer.offset(64));
+                r[5].store(buffer.offset(80));
+                r[6].store(buffer.offset(96));
+                r[7].store(buffer.offset(112));
+                buffer = buffer.offset(128);
+            }
+            remaining -= 128;
+        }
+        while remaining > 32 {
+            c[1] = c[0] + one;
+            compute_rounds!(0, c, r, s, self.round_keys);
+            compute_rounds!(1, c, r, s, self.round_keys);
+            c[0] = c[1] + one;
+            unsafe {
+                r[0].store(buffer.offset(0));
+                r[1].store(buffer.offset(16));
+                buffer = buffer.offset(32);
+            }
+            remaining -= 32;
+        }
+        while remaining > 16 {
+            compute_rounds!(0, c, r, s, self.round_keys);
+            c[0] = c[0] + one;
+            unsafe {
+                r[0].store(buffer);
+                buffer = buffer.offset(16);
+            }
+            remaining -= 16;
+        }
+        if remaining > 0 {
+            compute_rounds!(0, c, r, s, self.round_keys);
+            c[0] = c[0] + one;
+            unsafe {
+                #[repr(align(16))]
+                struct Aligned([u8; 16]);
+                let mut t: Aligned = std::mem::uninitialized();
+                let t = t.0.as_mut_ptr();
+                r[0].store(t);
```

Comment:
I'm not sure about this part. See [the original implementation](https://github.com/jedisct1/aes-stream/blob/9ff663848e1aa838e214b5b8df426b0ba614cc44/src/aes-stream.c#L80).